utils/scm/openebs/delete_cstor_csi_clone.yml

```diff
@@ -0,0 +1,115 @@
+  args:
+    executable: /bin/bash
+  register: status
+  failed_when: "status.rc ==0"
+
+## verify snapshot cr should not be deleted
+# - name: delete snapshot cr
+#   shell: >
+#     kubectl delete volumesnapshot.snapshot {{ snapshot_name }} -n {{ namespace }}
+#   args:
+#     executable: /bin/bash
+#   register: status
+#   failed_when: "status.rc ==0"
+
+## store the PV and CVRs for corresponding cloned pvc 
+- name: Derive PV name from PVC
+  shell: >
+    kubectl get pvc {{ cloned_pvc }} -n {{ namespace }}
+    --no-headers -o custom-columns=:spec.volumeName
+  args:
+    executable: /bin/bash
+  register: pv
+
+- set_fact:
+    cloned_pv_name: "{{ pv.stdout }}"
+
+- name: Obtaining the CVR from PV
+  shell: >
+    kubectl get cvr -n {{ operator_ns }} -l cstorvolume.openebs.io/name={{ pv.stdout }} 
+    --no-headers  -o custom-columns=:metadata.name
+  args:
+    executable: /bin/bash
+  register: cvr
+
+- set_fact:
+    cloned_cvr_name: "{{ cvr.stdout }}"
+
+## obtain the application pod name
+- name: Get the app pod name
+  shell: >
+    kubectl get pods -n {{ namespace }} -l app={{ app_label }} 
+    --no-headers -o custom-columns=:.metadata.name
+  args: 
+    executable: /bin/bash
+  register: app_pod_name
+
+- set_fact: 
+    application_podname: "{{ app_pod_name.stdout }}"
+
+## delete the busybox deployment  
+- name: delete the busybox_app 
+  shell: > 
+    kubectl delete deploy {{ app_name }} -n {{ namespace }}
+  args:
+    executable: /bin/bash
+  register: status
+  failed_when: "status.rc != 0"
+
+## verify if the pod is deleted
+- name: Check if the pod is deleted successfully
+  shell: >
+    kubectl get po -n {{ namespace }} 
+  args:
+    executable: /bin/bash
+  register: app_status
+  until: "application_podname not in app_status.stdout"
+  delay: 30
+  retries: 15   
+
+## verify and delete the cloned pvc 
+- name: delete the cloned pvc
+  shell: >
+    kubectl delete pvc {{ cloned_pvc }} -n {{ namespace }} 
+  args:
+    executable: /bin/bash
+  register: status
+  failed_when: "status.rc != 0"
+
+## verify if the cloned pvc is deleted
+- name: try to obtain the cloned pvc
+  shell: >
+    kubectl get pvc -n {{ namespace }}
+  args:
+    executable: /bin/bash
+  register: cloned_pvc_name
+  until: "cloned_pvc not in cloned_pvc_name  .stdout"
+  delay: 30
+  retries: 15   
+
+  ## verify if the pv for cloned pvc is deleted
+- name: try to obtain the pv for cloned pvc
+  shell: >
+    kubectl get pvc -n {{ namespace }}
+  args:
+    executable: /bin/bash
+  register: cloned_pv
+  failed_when: "cloned_pv_name in cloned_pv.stdout"
+
+  ## verify if the cvrs for cloned pvc is deleted
+- name: try to obtain the cvrs for cloned pvc
+  shell: >
+    kubectl get cvr -n {{ operator_ns }} -l cstorvolume.openebs.io/name={{ pv.stdout }} 
+    --no-headers  -o custom-columns=:metadata.name
+  args:
+    executable: /bin/bash
+  register: cloned_cvr
+  until: "cloned_cvr_name not in cloned_cvr.stdout"
```

Comment:
check this for volume with multiple replicas

Reply:
resolved